bcipy/helpers/task.py

```diff
@@ -370,186 +373,153 @@ def get_key_press(
-            # second dimension is trials and third dimension is time samples.
-            reshaped_trials = np.zeros(
-                (len(eeg_data), len(triggers), num_samples))
-
-            # Label for every trial
-            labels = np.zeros(len(triggers))
-
-            for trial in range(len(triggers)):
-                # Assign targetness to labels for each trial
-                if trial_target_info[trial] == 'target':
-                    labels[trial] = 1
-
-                # For every channel append filtered channel data to trials
-                for channel in range(len(eeg_data)):
-                    reshaped_trials[channel][trial] = \
-                        eeg_data[channel][
-                        triggers[trial]:triggers[trial] + num_samples]
-
-            # In copy phrase, num of inquiry is assumed to be 1.
-            num_of_inquiries = 1
-            # Since there is only one inquiry, all trials are in the inquiry
-            trials_per_inq = len(triggers)
-
-        # MODE FREE SPELL
-        elif mode == 'free_spell':
-
-            # triggers in sample are mapped to triggers in number of filtered
-            # samples.
-            triggers = list(
-                map(lambda x: int((x + offset) * after_filter_frequency), timing_info))
-
-            # 3 dimensional np array first dimension is channels
-            # second dimension is trials and third dimension is time samples.
-            reshaped_trials = np.zeros(
-                (len(eeg_data), len(triggers), num_samples))
-
-            labels = None
-
-            for trial in range(len(triggers)):
-
-                # For every channel append filtered channel data to trials
-                for channel in range(len(eeg_data)):
-                    reshaped_trials[channel][trial] = \
-                        eeg_data[channel][
-                        triggers[trial]:triggers[trial] + num_samples]
-
-            # In copy phrase, num of inquiry is assumed to be 1.
-            num_of_inquiries = 1
-            # Since there is only one inquiry, all trials are in the inquiry
-            trials_per_inq = len(triggers)
-        else:
-            raise Exception(
-                'Trial_reshaper does not work in this operating mode.')
-
-        # Return our trials, labels and some useful information about the
-        # arrays
-        return reshaped_trials, labels, num_of_inquiries, trials_per_inq
+    # Remove the channels that we are not interested in
+    channels_to_remove = [idx for idx, value in enumerate(channel_map) if value == 0]
+    eeg_data = np.delete(eeg_data, channels_to_remove, axis=0)
 
-    except Exception as e:
-        raise Exception(
-            f'Could not reshape trial for mode: {mode}, {fs}, {k}. Error: {e}')
+    # Number of samples we are interested per trial
+    num_samples = int(trial_length * fs)
+
+    # Remove unwanted elements from target info and timing info
+    targets_included = set(["target", "nontarget"])
```

Comment:
I think the same here as above in regards to accepting it as an argument. Ask for a list of things to include and potentially a targetness argument, which could be used ln 516.